src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs

```diff
@@ -51,11 +51,12 @@ public interface IFileStore
     /// <summary>
     /// Asynchronously get file properties
     /// </summary>
-    /// <param name="version">The DICOM instance version.</param>
-    /// <param name="partitionName">Name of the partition</param>
+    /// <param name="version">The DICOM instance version when file properties not known.</param>
+    /// <param name="partition">Partition of the instance to get file properties on when file properties not known</param>
+    /// <param name="fileProperties">When file properties known, will use to get content length and match on etag</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>A task that represents the asynchronous get properties operation.</returns>
-    Task<FileProperties> GetFilePropertiesAsync(long version, string partitionName, CancellationToken cancellationToken = default);
+    Task<FileProperties> GetFilePropertiesAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken = default);
```

Comment:
This hurts my brain to pass file properties for fetching file properties. For GA that we will start including content length in the SQL index?